### PR TITLE
Operation cancel in CachingTransport logs instead of errors

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Aims to solve: https://github.com/getsentry/sentry-dotnet/issues/1351

### CODE REVIEW

File: src/Sentry/Internal/Http/CachingTransport.cs [hunk 1]
```diff
@@ -158,7 +158,17 @@ private async Task ProcessCacheAsync(CancellationToken cancellationToken = defau
                 {
                     await InnerProcessCacheAsync(cancellationToken, envelopeFilePath).ConfigureAwait(false);
                 }
-                catch (Exception ex) when (IsRetryable(ex))
+                catch (OperationCanceledException ex) // OperationCancel should not log an error
+                {
+                    _options.LogDebug(
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Http/CachingTransport.cs:163]: Not sure we should retry. Maybe just swallow it here?

**Review by bruno-garcia — COMMENTED** [review]
Oh, forgot to send the review we did on a call

**Comment by bruno-garcia** [comment]
@bitsandfoxes please note a user raised an issue #1366 exactly about what we discussed during the review yesterday.

File: src/Sentry/Internal/Http/CachingTransport.cs [hunk 2]
```diff
@@ -158,7 +158,17 @@ private async Task ProcessCacheAsync(CancellationToken cancellationToken = defau
...
+                        "Canceled sending cached envelope: {0}, retrying after a delay.",
+                        ex,
+                        envelopeFilePath);
+
+                    // Let the worker catch, log, wait a bit and retry.
+                    throw;
+                }
```
[the remark this answers is not in the document]
Comment (SimonCropp) [reply src/Sentry/Internal/Http/CachingTransport.cs:174]: i added `_workerCts.IsCancellationRequested`. and tested a httptimeout, it threw a TaskCanceledException

**Comment by lucas-zimerman** [comment]
> ```gfm
> - Operation cancel in CachingTransport logs instead of errors ([#1352](https://github.com/getsentry/sentry-dotnet/pull/1352))
> ```

Isn't this wordy? perhaps 
```
- Decrease noise from canceled operations ([#1352](https://github.com/getsentry/sentry-dotnet/pull/1352))
```

What do you think @bruno-garcia?

**Comment by bruno-garcia** [comment]
> > ```
> > - Operation cancel in CachingTransport logs instead of errors ([#1352](https://github.com/getsentry/sentry-dotnet/pull/1352))
> > ```
> 
> Isn't this wordy? perhaps
> 
> ```
> - Decrease noise from canceled operations ([#1352](https://github.com/getsentry/sentry-dotnet/pull/1352))
> ```
> 
> What do you think @bruno-garcia?

I actually rather the changelog to be more descriptive. Thinking of the use case of it: A user wants to decide whether it's worth bumping the SDK version. They read the changelog. So the goal is to help them understand what we changed and if that resolved their issues or if they need to raise an issue with us, for example.

So the fact the name of the class is pointed out, is a good thing, IMHO

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
@bitsandfoxes @SimonCropp can we merge this? I think it only misses the changelog entry?

**Review by lucas-zimerman — APPROVED** [review]

**Pull request merged**